src/app/services/search.service.ts

```diff
@@ -0,0 +1,120 @@
+import {HttpClient, HttpParams} from "@angular/common/http";
+import {environment} from "../../environments/environment";
+import {Observable} from "rxjs";
+import {Injectable} from "@angular/core";
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SearchService {
+
+  private readonly baseUrl: string;
+  private _search: string;
+  private _geolocation: string;
+  private _sortType: string;
+  private _sortValue: string;
+  private _stars: number;
+  private _price: number;
+  private _page: number;
+  private _size: number;
+
+  constructor(private http: HttpClient) {
+
+    this._search = "";
+    this._geolocation = "";
+    this._sortType = "";
+    this._sortValue = "";
+    this._stars = 0;
+    this._price = 5;
+    this._page = 0;
+    this._size = 10;
+    if (!environment.production) {
+      this.baseUrl = 'http://localhost:8080';
+    }
+    else {
+      this.baseUrl = 'server url'; //TODO add server url when setup
+    }
+  }
+
+  searchRestaurants(): Observable<any> {
```

Comment:
This should be refactored if another page needs to use it, maybe a builder pattern. Not high priority though